.github/workflows/CI-windows.yml

```diff
@@ -26,79 +27,100 @@ jobs:
+          7z x z3-%Z3_VERSION%-win.zip -oexternals -r -y || exit /b !errorlevel!
+          move externals\z3-%Z3_VERSION%-${{ matrix.arch }}-win externals\z3 || exit /b !errorlevel!
         env:
           # see https://github.com/Z3Prover/z3/releases:  
           Z3_VERSION: 4.8.10
 
+      # no 32-bit Qt available
       - name: Install Qt
+        if: matrix.os == 'x64'
         uses: jurplel/install-qt-action@v2
         with:
           modules: 'qtcharts'
 
-      - name: Build GUI x64 release
+      - name: Build GUI release
+        if: matrix.os == 'x64'
         run: |
-          call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvars64.bat" 
-          cd gui
-          qmake HAVE_QCHART=yes
-          nmake release
+          call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" ${{ matrix.arch }} || exit /b !errorlevel!
+          cd gui || exit /b !errorlevel!
+          qmake HAVE_QCHART=yes || exit /b !errorlevel!
+          nmake release || exit /b !errorlevel!
       
       - name: Deploy GUI
+        if: matrix.os == 'x64'
         run: |
-          windeployqt Build\gui
-          del Build\gui\cppcheck-gui.ilk
-          del Build\gui\cppcheck-gui.pdb
+          windeployqt Build\gui || exit /b !errorlevel!
+          del Build\gui\cppcheck-gui.ilk || exit /b !errorlevel!
+          del Build\gui\cppcheck-gui.pdb || exit /b !errorlevel!
           
-      - name: Build CLI x64 debug configuration using MSBuild
-        run: msbuild -m cppcheck.sln /p:Configuration=Debug-PCRE /p:Platform=x64
+      - name: Build CLI debug configuration using MSBuild
+        run: |
+          call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" ${{ matrix.arch }} || exit /b !errorlevel!
+          set ARCH=${{ matrix.arch }}
+          if "${{ matrix.arch }}" == "x86" (
+            set ARCH=Win32
+          )
+          msbuild -m cppcheck.sln /p:Configuration=Debug-PCRE;Platform=%ARCH% -maxcpucount || exit /b !errorlevel!
 
       - name: Run Debug test
-        run: .\bin\debug\testrunner.exe
+        run: .\bin\debug\testrunner.exe || exit /b !errorlevel!
 
-      - name: Build x64 release configuration using MSBuild
-        run: msbuild -m cppcheck.sln /p:Configuration=Release-PCRE /p:Platform=x64
+      - name: Build CLI release configuration using MSBuild
+        run: |
+          call "C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Auxiliary\Build\vcvarsall.bat" ${{ matrix.arch }} || exit /b !errorlevel!
+          set ARCH=${{ matrix.arch }}
+          if "${{ matrix.arch }}" == "x86" (
+            set ARCH=Win32
+          )
+          msbuild -m cppcheck.sln /p:Configuration=Release-PCRE;Platform=%ARCH% -maxcpucount || exit /b !errorlevel!
 
       - name: Run Release test
-        run: .\bin\testrunner.exe
+        run: .\bin\testrunner.exe || exit /b !errorlevel!
```

Comment:
what does this exit accomplish? if testrunner succeed why do we need that extra stuff..

Reply:
I wasn't sure if the last `%ERRORLEVEL%` in a batch script is propagated to the caller. I just tested it and if there is just a single command we don't need to do this. Actually we don't need to do it for the last one either but in case someone extends the steps he might miss adding it. So I think it would be fine to keep it.

Reply:
hmm.. so you are saying that if there are two commands and the first fail and the second succeed.. then this script will succeed. that was unexpected imho. I'd expect an immediate abort.

Reply:
A bash script behaves the same unless you specify `set -e` in it. Batch scripts unfortunately do not have such a "fail all" option.

Reply:
I don't know if this is a batch script. Could we use https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idstrategyfail-fast ?

Reply:
It definitely is.
`fail-fast` has nothing to do with the steps but with the jobs. So in this if the x86 build fails it will immediately cancel the x64 one as well.

Reply:
hmm ok.. thanks for the clarification. I am surprised that this CI script might not fail when a step is failing..